dbms/src/Core/SettingsCommon.cpp

```diff
@@ -552,9 +546,10 @@ void SettingChar::set(char x)
 
 void SettingChar::set(const String & x)
 {
-    checkStringIsACharacter(x);
-    value = x[0];
-    changed = true;
+    if (x.size() > 1)
```

Comment:
Minor change to allow setting `SettingChar` to `'\0'` just specifying `''`.
For example,
```
create table simple (
  t UInt64,
  url String
) ENGINE=Kafka SETTINGS kafka_broker_list = 'localhost:9092',
                        kafka_topic_list = 'topic',
                        kafka_group_name = 'mygroup',
                        kafka_format = 'Protobuf',
                        kafka_row_delimiter = '',
                        kafka_schema = 'simple:AccessLog';
```